cpu/atmega_common/periph/gpio_ll_irq.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (C) 2014-2015 Freie Universität Berlin
+ *               2015 Hamburg University of Applied Sciences
+ *               2017-2020 Inria
+ *               2017 OTA keys S.A.
+ *               2021 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_stm32
+ * @ingroup     drivers_periph_gpio_ll_irq
+ * @{
+ *
+ * @file
+ * @brief       Low-level GPIO-NG IRQ driver implementation for STM32 (except F1)
```

Comment:
My idea was to start with the STM32 code and adapt it to the ATmega periphal, so that I have all the function signatures in place. I didn't intent to commit that while it is not yet finished, but apparently I did :)